tests/functional/lib/receptorcontrol/receptorcontrol.go

```diff
@@ -377,8 +378,17 @@ func (r *ReceptorControl) AssertWorkReleased(ctx context.Context, unitID string)
 }
 
 func (r *ReceptorControl) getWorkResults(unitID string, readSize int) ([]byte, error) {
-	r.WriteStr(fmt.Sprintf("work results %s\n", unitID))
-	r.ReadStr() // flush "Streaming results for.." line"
+	_, err := r.WriteStr(fmt.Sprintf("work results %s\n", unitID))
+	if err != nil {
+		return nil, err
+	}
+	str, err := r.ReadStr()
+	if err != nil {
+		return nil, err
+	}
+	if str[:5] == "ERROR" {
```

Comment:
Tests should always check error returns, particularly in obscure places where we don't expect them.  Not checking this particular error made the test hang (since the server side isn't going to send anything for the next `Read()` to read), and also made it a lot harder to figure out what the actual problem was in the tests.

I've added checks to several error returns in this file.  It might be worth looking through the rest of the tests to see if we have other cases like this.

Reply:
okay I'll comb through the workceptor tests and make sure we return errors